app/services/provider_interface/send_organisation_permissions_emails.rb

```diff
@@ -0,0 +1,53 @@
+module ProviderInterface
+  class SendOrganisationPermissionsEmails
+    def initialize(provider_user:, permissions:, provider: nil, set_up: false)
+      @provider_user = provider_user
+      @permissions = permissions
+      @provider = provider
+      @set_up = set_up
+    end
+
+    def call
+      managing_users.uniq.each do |user|
+        ProviderMailer.send("organisation_permissions_#{email_to_send}", user, partner_organisation, @permissions).deliver_later
+      end
+    end
+
+  private
+
+    attr_reader :permissions, :provider, :provider_user, :set_up
+
+    def managing_users
+      ProviderUser.joins(:provider_permissions)
+        .where(ProviderPermissions.table_name => { provider: partner_organisation, manage_organisations: true })
+    end
+
+    def email_to_send
+      set_up ? 'set_up' : 'updated'
+    end
+
+    def partner_organisation
+      if provider.present?
```

Comment:
My understanding of this is that we will only email the partner org users with manage org permissions, unless a user making the change belongs to both orgs, where we choose the first one alphabetically as the partner org (only when setting up orgs as its done outside of a context). I wonder if we can use the `provider` as a flag for set up vs updated, as we only set that when updating permissions?

Reply:
Yes that's a good point, the provider will always be `nil` coming from set-up. :+1: